src/kontrol/prove.py

```diff
@@ -468,9 +469,22 @@ def _method_to_cfg(
                 f'Initial state proof {setup_proof.id} for {contract.name_with_path}.{method.signature} still has pending branches.'
             )
 
-        init_node_id = setup_proof.init
+        if setup_proof.bounded:
+            raise RuntimeError(
+                f'Initial state proof {setup_proof.id} for {contract.name_with_path}.{method.signature} still has bounded branches.'
+            )
+
+        if setup_proof.failing:
+            raise RuntimeError(
+                f'Initial state proof {setup_proof.id} for {contract.name_with_path}.{method.signature} still has failing branches.'
+            )
 
-        cfg = KCFG.from_dict(setup_proof.kcfg.to_dict())  # Copy KCFG
+        assert setup_proof.status == ProofStatus.PASSED
+
+        init_node_id = setup_proof.init
+        # Copy KCFG and minimize it
+        cfg = KCFG.from_dict(setup_proof.kcfg.to_dict())
+        cfg.minimize()
```

Comment:
By minimizing the `setUp` KCFG, we reduce it to a single transition (unless `setUp` branches).